### DISCUSSION TITLE
Can you access methods directly?

### DISCUSSION THREAD

**Discussion by ZingBlue** [coqui-ai/TTS#603] [General Q&A] [answered]
Hello,

I'm using the TTS library as part of a project and after importing I noticed there were very few methods to auto-fill in my IDE.
After further research, I found that in server.py of the demo server, text is generated through CLI parsing.
I do not see any documentation for directly accessing methods in order to create audio from text;
this leads to the question: Is it possible to access methods for audio synthesis directly, or is parsing CLI arguments necessary?
And If CLI parsing is necessary, why?

Thanks in advance

**Comment by erogol** [comment]
you can access the methods as you like by importing the module. 

The latest docs might help https://tts.readthedocs.io/en/dev/

**Reply by ZingBlue** [reply]
For some reason the TTS module does not include any methods except `f`, `os`, `version`, and a few variables. I'm not sure why, I installed using the instructions GuyPaddock provided for local installation on Windows. I can use CLI commands such as `tss --text "Hello."` however.

**Reply by ZingBlue** [reply]
I tried installing via Pip for Python 3.8, same issue. I can use a few methods such as `version` (which accurately returns the version), but the majority of methods and classes such as util are unattainable. I checked the TTS folder in site packages, the `utils`, `vocoder` and `tts` all exist, I don't know why it is not working.

**Reply by erogol** [reply]
try installing from github by cloning it. You can find our how in the readme.

**Reply by erogol** [reply]
pip is mostly for just running inference

**Reply by ZingBlue** [reply]
Yeah I initially cloned the repo and installed using the instructions in the readme, it did not work however. Then I installed via pip using Python 3.8, and had the same issue.

**Reply by erogol** [reply]
didn't work means couldn't install? If so what was the error ?

**Reply by ZingBlue** [reply]
I followed the instructions and it installed without error, however afterwards I had to manually move the TTS subfolder into site-packages. Afterwards when I imported TTS into Python it only allowed me to access a few methods.

**Reply by erogol** [reply]
I am not sure what is wrong in your case. Please file a bug report.

**Comment by ZingBlue** [comment, marked as answer]
All further discussion occurs at issue #633.
